docs/book/user-guide/tutorial/trigger-pipelines-from-external-systems.md

```diff
@@ -0,0 +1,881 @@
+
+# Simple API key authentication
+# This environment variable serves as a security token to protect your API endpoints
+# In production, use a strong, randomly generated key stored securely
+API_KEY = os.environ.get("PIPELINE_API_KEY", "your-secure-api-key")
+api_key_header = APIKeyHeader(name="X-API-Key")
+
+async def get_api_key(api_key: str = Security(api_key_header)):
+    if api_key != API_KEY:
+        raise HTTPException(status_code=401, detail="Invalid API key")
+    return api_key
+
+# Request model for pipeline parameters
+class StepParameter(BaseModel):
+    parameters: Dict[str, Any]
+
+class PipelineRequest(BaseModel):
+    pipeline_name: str
+    steps: Dict[str, StepParameter] = {}
+    config_path: Optional[str] = None
+
+# Import a pipeline dynamically
+def import_pipeline(pipeline_name):
+    """Import a pipeline function from available modules."""
+    # First try to import from known pipelines
+    if pipeline_name == "training_pipeline":
+        return training_pipeline
+    
+    # Try importing from other modules
+    try:
+        spec = importlib.util.find_spec("common")
+        if spec is None:
+            raise ImportError(f"Module 'common' not found")
+            
+        module = importlib.util.module_from_spec(spec)
+        spec.loader.exec_module(module)
+        
+        if not hasattr(module, pipeline_name):
+            raise AttributeError(f"Pipeline '{pipeline_name}' not found in module")
+            
+        return getattr(module, pipeline_name)
+    except Exception as e:
+        raise HTTPException(status_code=404, detail=f"Pipeline not found: {str(e)}")
+
+@app.post("/trigger")
+async def trigger_pipeline(
+    request: PipelineRequest, 
+    api_key: str = Depends(get_api_key)
+):
+    """Trigger a ZenML pipeline with the given parameters."""
+    try:
+        # Dynamically import the pipeline
+        pipeline_func = import_pipeline(request.pipeline_name)
+        
+        # Configure the pipeline
+        if request.config_path:
+            configured_pipeline = pipeline_func.with_options(config_path=request.config_path)
+        else:
+            configured_pipeline = pipeline_func
+        
+        # Extract parameters from steps
+        step_parameters = {}
+        if request.steps:
+            for step_name, step_config in request.steps.items():
+                if step_config.parameters:
+                    step_parameters.update(step_config.parameters)
+        
+        # Run the pipeline with the provided parameters
+        run = configured_pipeline(**step_parameters)
+        
+        return {
+            "status": "success", 
+            "message": f"Pipeline '{request.pipeline_name}' triggered successfully",
+            "run_id": run.id if hasattr(run, 'id') else None
+        }
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"Failed to trigger pipeline: {str(e)}")
+
+@app.post("/trigger-async")
```

Comment:
i mean all things are async no? just do /trigger?

Reply:
The difference is that when you curl with `trigger-async` it hangs on the user end until everything is complete.

`trigger-async` it just immediately gives you the 200 that the pipeline was triggered but doesn't wait. I guess you want only the trigger-async version that doesn't synchronously hold until completion?

Reply:
Would you prefer I delete the synchronous endpoint completely and rename `trigger-async` to just `trigger`? I guess it makes it more of an apples to apples comparison with run templates triggering...

Reply:
Fixed.